Libraries/dotNetRdf.LinkedDataFragments/Parameters.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+// <copyright>
+// dotNetRDF is free and open source software licensed under the MIT License
+// -------------------------------------------------------------------------
+// 
+// Copyright (c) 2009-2023 dotNetRDF Project (http://dotnetrdf.org/)
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is furnished
+// to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// </copyright>
+*/
+
+using System;
+using System.Collections.Generic;
+using Resta.UriTemplates;
+using VDS.RDF.LinkedPatternFragments.Hydra;
+using VDS.RDF.Writing.Formatting;
+
+namespace VDS.RDF.LinkedPatternFragments
+{
+    internal class Parameters
+    {
+        private readonly IriTemplate template;
+        private readonly INode subject;
+        private readonly INode predicate;
+        private readonly INode @object;
+
+        internal Parameters(IriTemplate template, INode subject = null, INode predicate = null, INode @object = null)
+        {
+            if (template is null)
+            {
+                throw new ArgumentNullException(nameof(template));
+            }
+
+            if (subject is object && subject.NodeType != NodeType.Uri)
```

Comment:
I think ` != null` is maybe clearer to read than `is object` ?

Reply:
173a0bd